src/checkers/inference/model/serialization/JsonSerializer.java

```diff
@@ -329,6 +336,22 @@ public JSONObject serialize(PreferenceConstraint constraint) {
         return obj;
     }
 
+    @Override
+    public JSONObject serialize(ImplicationConstraint implicationConstraint) {
+        if (implicationConstraint.getAssumptions() == null || implicationConstraint.getConclusion() == null) {
```

Comment:
wouldn't it be impossible for these 2 to be null for any created implication constraints?

Reply:
Done